app/models/manageiq/providers/amazon/cloud_manager/refresh_worker.rb

```diff
@@ -5,11 +5,16 @@ class ManageIQ::Providers::Amazon::CloudManager::RefreshWorker < ManageIQ::Provi
   # Amazon-manager types from here.
   # This way, the refresher for Amazon's CloudManager will refresh *all*
   # of the Amazon inventory across all managers.
-  def self.queue_name_for_ems(ems)
-    if ems.kind_of?(ExtManagementSystem)
-      ["ems_#{ems.id}"] + ems.child_managers.collect { |manager| "ems_#{manager.id}" }
-    else
-      super
+  class << self
+    def queue_name_for_ems(ems)
+      return ems unless ems.kind_of?(ExtManagementSystem)
+      combined_managers(ems).collect(&:queue_name).sort
```

Comment:
Ah cool, so `&:queue_name` is going to return `ems_{ems.id}` I assume.

Reply:
Yessir https://github.com/ManageIQ/manageiq/blob/master/app/models/ext_management_system.rb#L487-L489 :)

Reply:
Also I got rid of the `super` because I got rid of the other case https://github.com/ManageIQ/manageiq/pull/16437 `return "generic" if ems.kind_of?(Host) && ems.acts_as_ems?`

Reply:
Super :)

Reply:
I'm going to try to get rid of the ems_or_queue_name thing also so `queue_name_for_ems` _actually_ is passed an ems.

Reply:
@agrare Nice, yeah I was looking last week at queues and workers. Love to chat to you about it sometime.